primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java

```diff
@@ -128,6 +145,24 @@ protected void encode(FacesContext context, Badge badge, BadgeModel badgeModel,
         }
     }
 
+    protected void encodeIcon(FacesContext context, Badge badge, String icon) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        String iconStyleClass = getStyleClassBuilder(context)
+                .add(Badge.ICON_CLASS)
+                .add(icon)
+                .build();
+        writer.startElement("i", null);
```

Comment:
Accessibility-wise you should use semantic tags. `i` is deprecated and `em` should be used instead. However, that's not semantic in this case. Something like `span` should be used instead. https://www.w3.org/TR/WCAG20-TECHS/G115.html